plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -825,6 +833,110 @@ public List<SourceFile> parse(Project subproject, Set<Path> alreadyParsed, Execu
+                    implementationClasspath = rewriteImplementation.resolve();
+                } catch (Exception e) {
+                    logger.warn("Failed to resolve dependencies from {}:{}. Some type information may be incomplete",
+                            subproject.getPath(), implementationName);
+                    implementationClasspath = emptySet();
+                }
+
+                String compileName = (String) sourceSet.getClass().getMethod("getCompileOnlyConfigurationName").invoke(sourceSet);
+                Configuration compileOnly = subproject.getConfigurations().getByName(compileName);
+                Configuration rewriteCompileOnly = subproject.getConfigurations().maybeCreate("rewrite" + compileName);
+                rewriteCompileOnly.setCanBeResolved(true);
+                rewriteCompileOnly.extendsFrom(compileOnly);
+
+                // The implementation configuration doesn't include build/source directories from project dependencies
+                // So mash it and our rewriteImplementation together to get everything
+                List<Path> dependencyPaths = Stream.concat(implementationClasspath.stream(), rewriteCompileOnly.getFiles().stream())
+                        .map(File::toPath)
+                        .map(Path::toAbsolutePath)
+                        .map(Path::normalize)
+                        .distinct()
+                        .collect(toList());
+
+                if (!kotlinPaths.isEmpty()) {
+                    logger.info("Parsing {} Kotlin sources from {}/{}", kotlinPaths.size(), subproject.getName(), kotlinDirectorySet.getName());
+                    KotlinParser kp = KotlinParser.builder()
+                            .classpath(dependencyPaths)
+                            .styles(getStyles())
+                            .typeCache(javaTypeCache)
+                            .logCompilationWarningsAndErrors(extension.getLogCompilationWarningsAndErrors())
+                            .build();
+
+                    Instant start = Instant.now();
+                    List<K.CompilationUnit> cus = kp.parse(kotlinPaths, baseDir, ctx);
+                    alreadyParsed.addAll(kotlinPaths);
+                    cus = ListUtils.map(cus, cu -> {
+                        if (isExcluded(exclusions, cu.getSourcePath())) {
+                            return null;
+                        }
+                        return cu;
+                    });
+                    Duration parseDuration = Duration.between(start, Instant.now());
+                    logger.info("Finished parsing Kotlin sources from {}/{} in {} ({} per source)",
+                            subproject.getName(), kotlinDirectorySet.getName(), prettyPrint(parseDuration), prettyPrint(parseDuration.dividedBy(kotlinPaths.size())));
+                    List<SourceFile> sourceFiles = new ArrayList<>(cus.size());
+                    sourceFiles.addAll(map(autodetectStyle(cus), addProvenance(emptyList(), null)));
```

Comment:
Might be a good idea to add a `JavaSourceSet` marker. Even if you don't know all the types just having the name of the source set will be important for some recipes.

Reply:
Resolved by: https://github.com/openrewrite/rewrite-kotlin/commit/ff98528d18db1959d29b51bf6d5936890c6eac9e